src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -99,9 +99,21 @@ static List<String> getFieldNamesFromProjectStage(BsonDocument projectStage) {
         return fieldNames;
     }
 
-    private static boolean trueOrOne(BsonValue value) {
-        return (value.isBoolean() && value.asBoolean().getValue())
-                || (value.isNumber() && value.asNumber().intValue() == 1);
+    private static boolean isExcludeProjectSpecification(Map.Entry<String, BsonValue> specification) {
+        var key = specification.getKey();
+        var value = specification.getValue();
+        var exclude = (value.isBoolean() && !value.asBoolean().getValue())
+                || (value.isNumber() && value.asNumber().intValue() == 0);
+        if (exclude && !key.equals(ID_FIELD_NAME)) {
+            throw new RuntimeException(format(
+                    "Exclusions are not allowed in `$project` specifications, except for the [%s] field: [%s, %s]",
+                    ID_FIELD_NAME, key, value));
+        }
+        if (!value.isBoolean() && !value.isInt32()) {
+            throw new FeatureNotSupportedException(format(
+                    "Expressions and literals are not supported in `$project` specifications: [%s: %s]", key, value));
```

Comment:
This is likely to change. When it does, we should try to allow only the bare minimum we need or know works (for example, because we have tests), and keep forbidding the rest.